### ISSUE TITLE
Is it possible to use direct references instead of lifetime-parameterized structs?

### ISSUE THREAD

**Issue by shepmaster** [jpernst/rental#20]
Specifically, I mean something like:

```rust
#[rental]
pub struct MyRental {
    greeting: String,
    greeting_1: &'greeting str,
    greeting_2: &'greeting str,
}
```

Currently, this fails with 

```
   = help: message: Field `greeting_1` must be a type path.
```

Is this a unsolvable problem or just a not-yet-implemented feature?

**Comment by jpernst** [comment]
This is probably just an oversight on my part, I'll look into it.

**Comment by jpernst** [comment]
Ok, I've identified the problem. It should be fixable, but as a workaround in the meantime you can put the two string slices in a tuple field instead of as two separate fields. I'd actually recommend doing that anyway since `greeting_2` does not depend on `greeting_1` at all, so they can be constructed simultaneously. In rental structs, each field has an implicit dependency on previous fields, so using as few fields as you can get away with will make it easier to use.

**Comment by jpernst** [comment]
Fixed in 0.4.14
